### PR TITLE
Fix uncaught exception when trying to retrieve negative addresses

### PR DESCRIPTION
**Pull request by TETYYS**

### CODE REVIEW

**Comment by KN4CK3R** [comment]
Did you test a negative address intentional or how did you get this exception? Your patch doesn't fix the problem. Now you can't access offsets greater than int.MaxValue (2GB). I'm a little bit shocked that y and z are both negative...

```
var x = new IntPtr(-1);
var y = x.ToInt32();
var z = x.ToInt64();
```

I expected z to show a large number instead of -1.

**Comment by TETYYS** [comment]
I think I got the exception when the program tried to construct a BaseHexCommentNode, presumably dereferencing an address, my file is exactly 2,147,516,415 bytes in size.

**Comment by TETYYS** [comment]
I think the problem with this is that when pointer in IntPtr is 0x8000_0000, the ToInt64() operation does not cast bit-wise, resulting in same negative value

**Comment by TETYYS** [comment]
I think this should work. The plugin fails to read valid bytes if you try to read after the file, but that's a different issue, I think I saw 64 byte blocks hardcoded somewhere, this might be related to that.

File: LoadBinaryPluginExt.cs [hunk 1]
```diff
@@ -164,12 +164,6 @@ public bool ReadRemoteMemory(IntPtr process, IntPtr address, ref byte[] buffer,
 						{
 							addressInt64 &= 0xFFFF_FFFF;
 						}
-						using (var stream = info.File.CreateViewStream(addressInt64, size))
-						{
-							stream.Read(buffer, 0, size);
-
-							return true;
-						}
 						using (var stream = info.File.CreateViewStream(address.ToInt64(), size))
```
Comment (KN4CK3R) [inline LoadBinaryPluginExt.cs]: Now you are missing the addressInt64 here.

**Comment by TETYYS** [comment]
There are some places in C# library methods I really wish that would just accept or provide `unsigned` values.

**Comment by KN4CK3R** [comment]
I will add a new extension method for IntPtr in the ReClass.NET project to make the logic reusable.

**Pull request merged**